samcli/lib/telemetry/event.py

```diff
@@ -195,10 +202,25 @@ def clear_trackers():
     @staticmethod
     def send_events() -> threading.Thread:
         """Call a thread to send the current list of Events via Telemetry."""
+        EventTracker._set_session_id()
+
         send_thread = threading.Thread(target=EventTracker._send_events_in_thread)
         send_thread.start()
         return send_thread
 
+    @staticmethod
+    def _set_session_id() -> None:
+        """
+        Get the session ID from click and save it locally.
+        """
+        if not EventTracker._session_id:
```

Comment:
I don't see why there would be, but wanted to confirm that there won't be any collisions or race-conditions between retrieving and setting the `session_id` from click since it was moved from a previously locked state out.

Reply:
This is a good catch. I can move the function call back to the locked state so that we don't accidentally undo anything that was caught before.